063-pdb-generation-environment-variable.md

```diff
@@ -0,0 +1,49 @@
+# Pod Disruption Budget Generation Environment Variable Proposal
+
+## Background and Problem Statement
+
+The Strimzi Kafka operator currently lacks the option to disable the creation of Pod Disruption Budgets (PDBs). In certain infrastructures, PDB creations for users are denied, resulting in operational challenges. Providing an option to disable PDB creation in Strimzi would address these constraints.
+
+## Proposed Solution
+
+Introduce a new environment variable in Strimzi, [similar to the one for Network Policies](https://github.com/strimzi/proposals/blob/main/028-network-policy-generation-environment-variable.md), that allows global disabling of PDB creation. This proposal suggests the environment variable `STRIMZI_POD_DISRUPTION_BUDGET_GENERATION`, which by default is `true`. When set to `false`, the Strimzi operator will not generate Pod Disruption Budgets.
+
+## Rationale
+
+- **Flexibility:** Some users operate in environments with strict policies that prevent the creation of PDBs. Providing an option to disable PDB generation caters to these scenarios.
+- **Consistency:** This approach is consistent with existing features, such as the Network Policy Generation Environment Variable.
+- **Simplicity:** A global setting avoids the complexity of configuring this setting for each custom resource.
+
+## Current Implementation
+
+The Strimzi operator automatically creates PDBs to ensure high availability and minimize disruptions. However, this feature does not accommodate environments where PDB creation is restricted.
+
+## Motivation
+
+- **Operational Constraints:** In environments where PDB creation is not permitted, users are unable to deploy Strimzi effectively.
+- **Configuration Flexibility:** Users need the ability to configure their Strimzi deployments to align with their specific operational policies and constraints.
+
+## Proposal Details
+
+1. **Environment Variable Introduction:** Implement `STRIMZI_POD_DISRUPTION_BUDGET_GENERATION` with a default value of `true`.
+2. **Disabling PDB Generation:** When set to `false`, the operator will skip PDB creation for all Strimzi components.
```

Comment:
Maybe you can clarify that it will not create or update them either?

```suggestion
2. **Disabling PDB Generation:** When set to `false`, the operator will skip PDB operations for all Strimzi components. It will not create, modify or delete any PDB.
```

This is useful  to make it clear that when you disable it in existing cluster, the operator will not attempt to delete any PDB etc.